src/getstream/updateBetterChannelMembers.js

```diff
@@ -78,13 +89,13 @@ const __helperProcessBetterChannelMember = (members, membersDataFromDbMap) => {
       : username;
 
     newChannelName += `${updatedUsername}${SEPARATOR}`;
-
+    member = removeDisabledDevice(member);
     const defaultUser = {
       ...member,
       is_anonymous,
       user: {
         ...member.user,
-        is_anonymous: is_anonymous,
+        is_anonymous,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `removeDisabledDevice` function is mutating the input parameter `member`. This can lead to unexpected side effects. It's better to avoid mutating function parameters directly. Instead, create a new object and return that.

```diff
- member = removeDisabledDevice(member);
+ const updatedMember = removeDisabledDevice(member);

- const defaultUser = {
-   ...member,
-   is_anonymous,
-   user: {
-     ...member.user,
-     is_anonymous,
-     username: updatedUsername,
-     name: updatedUsername
-   }

+ const defaultUser = {
+   ...updatedMember,
+   is_anonymous,
+   user: {
+     ...updatedMember.user,
+     is_anonymous,
+     username: updatedUsername,
+     name: updatedUsername
+   }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->